src/BoundaryConditions/flat_extrapolation_open_boundary_matching_scheme.jl

```diff
@@ -0,0 +1,89 @@
+using Oceananigans.Operators: ∂xᶜᶜᶜ
+
+"""
+    FlatExtrapolation
+
+Zero gradient perepndicular velocity boundary condition.
+
+*Given constant spacing*
+```math
+f′(xᵢ) ≈ f′(xᵢ₋₁) + f′′(xᵢ₋₁)(xᵢ₋₁ - xᵢ) + O(Δx²) = f′(xᵢ₋₁) + f′′(xᵢ₋₁)Δx + O(Δx²),
+
+f′(xᵢ₋₁) ≈ (f(xᵢ) - f(xᵢ₋₂)) / 2Δx
+
+f′(xᵢ) ≈ (f(xᵢ) - f(xᵢ₋₂)) / 2Δx + O(Δx) = 0 ∴ f(xᵢ) ≈ f(xᵢ₋₂) + O(Δx)
+
+f′′(xᵢ₋₁) ≈ (f′(xᵢ) - f′(xᵢ₋₂)) / 2Δx = - f′(xᵢ₋₂) / 2Δx ≈ - (f(xᵢ₋₁) - f(xᵢ₋₃)) / (2Δx)²
+
+∴ f(xᵢ) ≈ f(xᵢ₋₂) + (f(xᵢ₋₁) - f(xᵢ₋₃))/2 + O(Δx²)
+```
+"""
+struct FlatExtrapolation{FT}
+    relaxation_timescale :: FT
+end
+
+const FEOBC = BoundaryCondition{<:Open{<:FlatExtrapolation}}
+
+function FlatExtrapolationOpenBoundaryCondition(val = nothing; relaxation_timescale = Inf, kwargs...)
+    classifcation = Open(FlatExtrapolation(relaxation_timescale))
+    
+    return BoundaryCondition(classifcation, val; kwargs...)
+end
+
+@inline relax(j, k, c, bc, grid, clock, model_fields) =
+    c + ifelse(isnothing(bc.condition)||!isfinite(clock.last_stage_Δt), 0,
+        (getbc(bc, j, k, grid, clock, model_fields) - c) * min(1, clock.last_stage_Δt / bc.classification.matching_scheme.relaxation_timescale))
```

Comment:
I suggest changing the index names here just to avoid implying that this function only works for x-boundary conditions

```suggestion
@inline relax(l, m, c, bc, grid, clock, model_fields) =
    c + ifelse(isnothing(bc.condition)||!isfinite(clock.last_stage_Δt), 0,
        (getbc(bc, l, m, grid, clock, model_fields) - c) * min(1, clock.last_stage_Δt / bc.classification.matching_scheme.relaxation_timescale))
```